gobblin-core/src/main/java/gobblin/instrumented/writer/InstrumentedDataWriterDecorator.java

```diff
@@ -89,6 +89,15 @@ public long bytesWritten()
   }
 
   @Override
+  public State getFinalState() {
```

Comment:
If embedded writer implements FinalState, then we return the actual final state of the embedded object. However, DataWriter in general doesn't implement FinalState, so we return super in that case (which is just empty).